verilog/preprocessor/verilog_preprocess.cc

```diff
@@ -552,6 +555,116 @@ absl::Status VerilogPreprocess::HandleEndif(
   return absl::OkStatus();
 }
 
+// Handle `include directives.
+absl::Status VerilogPreprocess::HandleInclude(
+    TokenStreamView::const_iterator iter,
+    const StreamIteratorGenerator& generator) {
+  // TODO(karimtera): how to differentiate between <file> and "file"?? both are
+  // the same token, need to edit the lexer.
+  TokenStreamView::const_iterator token_iter = generator();
+  auto file_token_iter = *token_iter;
+  if (file_token_iter->token_enum() != TK_StringLiteral) {
+    preprocess_data_.errors.push_back(
+        {**token_iter, "Expected a path to a SV file."});
+    return absl::InvalidArgumentError("Expected a path to a SV file.");
+  }
+  // Currently the file path looks like "path", we need to remove "".
+  const auto& token_text = file_token_iter->text();
+
+  std::filesystem::path file_path =
+      std::string(token_text.substr(1, token_text.size() - 2));
+
+  std::string file_path_to_search = file_path.string();
+
+  if (file_path.is_relative()) {
+    bool found = false;
+    for (const auto& base_dir : preprocess_info_.include_dirs) {
+      file_path_to_search =
+          verible::file::JoinPath(base_dir, file_path.string());
+      auto search_status = verible::file::FileExists(file_path_to_search);
+      if (search_status.ok()) {
+        found = true;
+        break;
+      }
+    }
+    if (!found) {
+      preprocess_data_.errors.push_back(
+          {**token_iter,
+           absl::StrCat(file_path_to_search, ": no such file found.")});
+      return absl::InvalidArgumentError("ERROR: included file can't be found.");
+    }
+  } else {
+    if (auto status = verible::file::FileExists(file_path_to_search);
+        !status.ok()) {
+      preprocess_data_.errors.push_back(
+          {**token_iter, std::string(status.message())});
+      return absl::InvalidArgumentError("ERROR: included file can't be found.");
```

Comment:
instead of wrapping the status with a new one with a guess of what went wrong, woul'd it be better (_and_ easier) if we returned the original status that already contained all the most detailed issue ?
(similar in error cases below)